### PR TITLE
Enabling ObjC support for AliasTokens, plus expanded demo verification

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Exposed read-only access of `AliasTokens` to Objective-C clients.

Each `AliasTokens` property now has an equivalent Objective-C getter method. These methods are marked `obsoleted` for Swift clients, who should continue using the properties directly.

```swift
@available(swift, obsoleted: 1.0, message: "This method exists for Objective-C backwards compatibility and should not be invoked from Swift. Please use the `brandColors` property directly.")
@objc(brandColorFromToken:)
public func brandColor(_ token: BrandColorsTokens) -> DynamicColor {
    return brandColors[token]
}
```

This also required converting a number of Swift structs (`DynamicColors`, `ShadowInfo`, and `FontInfo`) to NSObject subclasses in order to be accessible from Objective-C. This change resulted in a pretty substantial binary size reduction, which is pretty cool!

Binary change:

| File | Before | After | Delta
| -- | -- | -- | -- |
| libFluentUI.a | 28,689,720 | 28,414,304 | -275,416 |
| FluentUITestApp | 31,734,955 | 31,691,301 | --43,654 |

### Verification

Expanded `ObjectiveCDemoController` to expose this code and test theme swapping.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 14 Pro - 2022-12-12 at 15 02 27 | Simulator Screen Recording - iPhone 14 Pro - 2022-12-12 at 15 01 37 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [x] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1452)

### CODE REVIEW

**Comment by mischreiber** [comment]
The ObjC method signatures in `AliasTokens` need a little work still (not to mention public comments) but it's a start to give the idea.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m [hunk 1]
```diff
@@ -40,18 +46,48 @@ - (void)viewDidLoad {
 
     MSFButton *testButton = [self createButtonWithTitle:@"Test" action:@selector(buttonPressed:)];
     [self.container addArrangedSubview:testButton];
+
+    [self setAddedLabels:[NSMutableSet set]];
+
+    self.appearanceController = [MSFDemoAppearanceControllerWrapper createDemoAppearanceControllerWithDelegate:nil];
```
Comment (jeaniehuynh) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m:52]: What's the reason for the mix-matched usage of dot notation/bracket notation throughout this file? (i.e. Why isn't this `[self setAppearanceController:[MSFDemoAppearanceControllerWrapper createDemoAppearanceControllerWithDelegate:nil]];`?)

File: ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift [hunk 2]
```diff
@@ -19,6 +21,13 @@ public final class AliasTokens {
         case tint30
         case tint40
     }
+
+    @available(swift, obsoleted: 1.0, message: "This method exists for Objective-C backwards compatibility and should not be invoked from Swift. Please use the `brandColors` property directly.")
+    @objc(brandColorFromToken:)
```
Comment (huwilkes) [inline ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift:26]: `FromToken` sounds odd to me. Would `ForToken` work?

File: ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift [hunk 3]
```diff
@@ -5,11 +5,13 @@
...
 import SwiftUI
 
-public final class AliasTokens {
+@objc(MSFAliasTokens)
+public final class AliasTokens: NSObject {
 
     // MARK: - BrandColors
 
-    public enum BrandColorsTokens: TokenSetKey {
+    @objc(MSFBrandColorsAliasTokens)
```
Comment (huwilkes) [inline ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift:13]: Why the extra alias in `MSFBrandColorsAliasTokens`?

**Review by huwilkes — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m [hunk 4]
```diff
@@ -40,18 +46,48 @@ - (void)viewDidLoad {
...
     [label setTextAlignment:NSTextAlignmentCenter];
-    [label setText:@"Test label with color"];
+    [label setText:text];
+    [label setTextColor:textColor];
+
+    [[self container] addArrangedSubview:label];
+    [[self addedLabels] addObject:label];
+}
 
+- (void)buttonPressed:(id)sender {
```
Comment (edjamesmsft) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m:79]: Is there a reason we're adding labels on a button press? This seems like an unnecessary step.

**Review by edjamesmsft — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/ObjectiveCDemoController.m:52]: Yeah, unfortunately the file is already a bit inconsistent. Let me see if I can do a tiny bit of cleanup without leaving the entire file as diffs.

File: ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift (hunk 2 above)
Comment (mischreiber) [reply ios/FluentUI/Core/Theme/Tokens/AliasTokens.swift:26]: Sounds good!

**Pull request merged**